### PR TITLE
get payments for tool to check backend status

### PR DESCRIPTION
**Pull request by thepiwo**

### CODE REVIEW

File: src/main/java/backend/controller/exceptions/SponsoringInvoiceController.kt [hunk 1]
```diff
@@ -37,7 +37,7 @@ class SponsoringInvoiceController(private val sponsoringInvoiceService: Sponsori
 
     @PostMapping("/sendmails")
     @PreAuthorize("hasAuthority('ADMIN')")
-            // TODO: Why does @Param work here?
+    // TODO: Why does @Param work here?
```
Comment (nerdsupremacist) [inline src/main/java/backend/controller/exceptions/SponsoringInvoiceController.kt:40]: Why shouldn't it work...

**Review by nerdsupremacist — APPROVED** [review]
lgtm

File: src/main/java/backend/controller/InvoiceController.kt [hunk 2]
```diff
@@ -104,6 +106,21 @@ class InvoiceController(private val teamEntryFeeService: TeamEntryFeeService,
...
     }
 
+    /**
+     * GET /invoice/payment/
+     * Allows admin to get all payments
+     */
+    @GetMapping("/payment/")
+    fun getPayments(@RequestHeader("X-AUTH-TOKEN") authToken: String): List<PaymentView> {
+
+        if (authToken != PAYMENT_AUTH_TOKEN) throw UnauthorizedException("Invalid Payment-Auth Token")
```
Comment (nerdsupremacist) [inline src/main/java/backend/controller/InvoiceController.kt:116]: I don't love this check though...

File: src/main/java/backend/controller/exceptions/SponsoringInvoiceController.kt (hunk 1 above)
Comment (thepiwo) [reply src/main/java/backend/controller/exceptions/SponsoringInvoiceController.kt:40]: Just formatting adjustment there, no idea about what this comment is about

File: src/main/java/backend/controller/InvoiceController.kt (hunk 2 above)
Comment (thepiwo) [reply src/main/java/backend/controller/InvoiceController.kt:116]: yeah, we should make a task to rebuild this using oauth for the payment tool

**Review by tgraupne — APPROVED** [review]

File: src/main/java/backend/controller/InvoiceController.kt (hunk 2 above)
Comment (florianschmidt1994) [reply src/main/java/backend/controller/InvoiceController.kt:116]: Can you open an issue for that?

**Review by florianschmidt1994 — APPROVED** [review]
LGTM

**Pull request merged**